### DISCUSSION TITLE
Base64 support by default for all File type serializers

### DISCUSSION THREAD

**Discussion by VaZark** [encode/django-rest-framework#8338] [General]
Given then that DRF is THE standard to use when creating REST APIs with Django, I've surprised that base64 support for file types has been pushed to third party libraries. Given that REST APIs almost always use base64 for files nowadays, would you be interested in adding an `encoding='base64'` as an option to the Image and FileSerializers?

**Comment by lovelydinosaur** [comment]
We pretty much suggest that *all* new functionality from this point ought to be in third party packages.
Still, it's a neat idea - what would the API look like to enable/disable it? What do responses look like both before and after the proposed change?

**Comment by VaZark** [comment]
This request is primarily targeted at allowing file upload when using `application/json`. At the moment, without third-party libraries file uploads are restricted to FormParser or FileUploadParser. Adding this would open a whole new set of possibilities as it'll allow endpoints to be purely JSON.

When active, the Base64Serializer should be exposed as string for OpenAPI but treated like a FileSerializer internally wrt the models.

I imagine we could extend the option with something like the following,
```python
FileField(max_length=None, allow_empty_file=False, use_url=UPLOADED_FILES_USE_URL, use_base64=True)
FileField(max_length=None, allow_empty_file=False, use_url=UPLOADED_FILES_USE_URL, encoding="base64")
```

**Reply by browser-bug** [reply]
At this day, how did you handle this? I'm trying to create a serializer that could manage a payload like the following:
```json
[
    {
        "file": "base64string",
        "label": "File label"
    },
    ...
]
```

**Reply by VaZark** [reply]
I just modified the flow as I realised I was in the wrong. base64 explodes the size of large files. I needed base64 as some of  the legacy code was on a no-code system that received only base64 instead of plain form-data

I offboarded and split my UI into a multistep form that receives all other data. I then hit the next endpoint to store the files.
